src/hooks/useFetch.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const getData = async (url) => {
+      const response = await fetch(url);
+
+      if (response.status !== 200) {
+        throw new Error (`${response.status} : Sorry, cannot fetch the data.`)
+      }
+      if (!response.ok) {
+        throw new Error ('Something has gone wrong, please try again.')
+      }
+
+      const data = await response.json();
+
+      return data;
+    }
+
+    getData()
+      .then(data => {
+        setData(data);
+        setIsLoading(false);
+        setError(null);
+      })
+      .catch(error => {
+        setError(error.message);
+        setIsLoading(false);
+      })
+  }, [url])
+
+  return { data, isLoading, error };
+}
+
+export default useFetch;
```

Comment:
What do you think about adding this load and error functionality without hooks? You could add the error  component to App.js state, pass it to the MainContainer.js and EmailNotificationForm.js files. 

We can figure out the loading component's use case once we have a large set of information that were waiting to retrieve from the BE in order to display on the FE (i.e. once we have something in App.js state).

Reply:
Neither one of these files are being used.  They were practice and then reference files that I will remove.